Add sparse QR solver functions (csrlsvqr) (#338)
ROCm/hipSOLVER: docs/howto/usage.rst

  * :ref:`hipsolverSpXcsrlsvcholHost <sparse_csrlsvcholHost>` with `reorder = 1`
  * :ref:`hipsolverSpXcsrlsvchol <sparse_csrlsvchol>` with `reorder = 1`

> - The functions :ref:`hipsolverSpScsrlsvqr <sparse_csrlsvqr>` are currently implemented by converting the sparse input matrix to a dense

Reviewer: ```suggestion
- The function :ref:`hipsolverSpScsrlsvqr <sparse_csrlsvqr>` is currently implemented by converting the sparse input matrix to a dense
```

  * :ref:`hipsolverSpXcsrlsvcholHost <sparse_csrlsvcholHost>` with `reorder = 1`
  * :ref:`hipsolverSpXcsrlsvchol <sparse_csrlsvchol>` with `reorder = 1`

- The function :ref:`hipsolverSpScsrlsvqr <sparse_csrlsvqr>` is currently implemented by converting the sparse input matrix to a dense

>   matrix, and therefore do not support any reordering method.

Reviewer 2: we should also mention that the HOST path is not supported at the moment

  (:ref:`hipsolverSpXcsrlsvchol <sparse_csrlsvchol>` may perform slower and will require more memory usage than
  :ref:`hipsolverSpXcsrlsvcholHost <sparse_csrlsvcholHost>`.)

> - The functions :ref:`hipsolverSpScsrlsvqr <sparse_csrlsvqr>` are currently implemented by converting the sparse input matrix to a dense
>   matrix, and then running the dense factorization and linear solver on the result. This may result in slower-than-expected performance and
>   significant memory usage for large matrices.

Reviewer: ```suggestion
- The function :ref:`hipsolverSpScsrlsvqr <sparse_csrlsvqr>` is currently implemented by converting the sparse input matrix to a dense
  matrix, and then running the dense factorization and linear solver on the result. This might result in slower-than-expected performance and
  significant memory usage for large matrices.
```
